packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js

```diff
@@ -372,8 +447,165 @@ async function askSecurityQuestions(context, parameters) {
+              value: 'AUTOMERGE',
+            },
+            {
+              name: 'Optimistic Concurrency',
+              value: 'OPTIMISTIC_CONCURRENCY',
+            },
+            {
+              name: 'Custom Lambda',
+              value: 'LAMBDA',
+            },
+            {
+              name: 'Learn More',
+              value: 'Learn More',
+            },
+          ],
+        };
+        ({ conflictResolutionStrategy } = await inquirer.prompt([conflictResolutionQuestion]));
+      } while (conflictResolutionStrategy === 'Learn More');
+
+      let syncConfig = {
+        ConflictHandler: conflictResolutionStrategy,
+        ConflictDetection: 'VERSION',
+      };
+
+      if (conflictResolutionStrategy === 'LAMBDA') {
+        const lambdaFunctionName = await askSyncFunctionQuestion(context);
+        syncConfig.LambdaConflictHandler = {};
+        syncConfig.LambdaConflictHandler.name = lambdaFunctionName;
+      }
+
+      return syncConfig;
+    };
+
+    resolverConfig.project = await askConflictResolutionStrategy('Select the default resolution strategy');
+
+    // Ask for per-model resolver override setting
+
+    if (modelTypes && modelTypes.length > 0) {
+      if (await context.prompt.confirm('Do you want to override default per model settings?')) {
+        const modelTypeQuestion = {
+          type: 'checkbox',
+          name: 'selectedModelTypes',
+          message: 'Select the models from below:',
+          choices: modelTypes,
+        };
+
+        const { selectedModelTypes } = await inquirer.prompt([modelTypeQuestion]);
+
+        if (selectedModelTypes.length > 0) {
+          resolverConfig.models = {};
+          for (let i = 0; i < selectedModelTypes.length; i += 1) {
+            resolverConfig.models[selectedModelTypes[i]] = await askConflictResolutionStrategy(
+              `Select the resolution strategy for ${selectedModelTypes[i]} model`
+            );
+          }
+        }
+      }
+    }
+  }
+
+  return resolverConfig;
+}
+
+async function askSyncFunctionQuestion(context) {
+  const syncLambdaQuestion = {
+    type: 'list',
+    name: 'syncLambdaAnswer',
+    message: 'Select from the options below',
+    choices: [
+      {
+        name: 'Create a new Lambda Function',
+        value: 'NEW',
+      },
+      {
+        name: 'Pre-existing Lambda Function',
+        value: 'EXISTING',
+      },
+    ],
+  };
+
+  const { syncLambdaAnswer } = await inquirer.prompt([syncLambdaQuestion]);
+
+  let lambdaFunctionName;
+
+  if (syncLambdaAnswer === 'NEW') {
+    lambdaFunctionName = await createSyncFunction(context);
+  } else {
+    const syncLambdaNameQuestion = {
+      type: 'input',
+      name: 'lambdaFunctionName',
+      message: 'Enter lambda function name',
+      validate: val => !!val,
+    };
+    ({ lambdaFunctionName } = await inquirer.prompt([syncLambdaNameQuestion]));
+  }
+
+  if (!lambdaFunctionName) {
+    throw new Error('Error in configuring Lambda function');
```

Comment:
The error message should be more specific that it is about the name not defined.

Reply:
I don't think we need this check at all as validation function won't let customer go to next question if they don't enter anything